5307 dataset creator default role api

sekmiller opened the pull request:
## New Contributors

Welcome! New contributors should at least glance at [CONTRIBUTING.md](/CONTRIBUTING.md), especially the section on pull requests where we encourage you to reach out to other developers before you start coding. Also, please note that we measure code coverage and prefer you write unit tests. Pull requests can still be reviewed without tests or completion of the checklist outlined below. Thanks!

## Related Issues

- connects to #5307 Add API to set default role for dataset creators

## Pull Request Checklist

- [x] Unit [tests][x] none
- [x] Integration [tests][x]: completed
- [x] Deployment requirements, [x] SQL Script
- [x] [Documentation][x] completed
- [x] Merged latest from "develop" [branch][x] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: http://guides.dataverse.org/en/latest/developers/sql-upgrade-scripts.html
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/7.3.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

pdurbin commented on doc/sphinx-guides/source/api/native-api.rst:
```diff
@@ -87,13 +87,34 @@ Create a New Role in a Dataverse
 Creates a new role under dataverse ``id``. Needs a json file with the role description::
 
   POST http://$SERVER/api/dataverses/$id/roles?key=$apiKey
+  
```
I find these "POST" examples that require a file to be sent across the wire to be terribly confusing unless we provide a full curl example like the `--upload-file facets.json` example above. While were in here, can we change it to be a full curl example? Should we also add a note saying that we don't really recommend creating custom roles and that you should consider opening a GitHub issue or starting a conversation on the mailing list if you are feeling the need to create a custom role?

pdurbin commented on src/main/java/Bundle.properties:
```diff
@@ -2029,6 +2029,7 @@ permission.roleWas=The role was {0}. To assign it to a user and/or group, click
 permission.roleNotSaved=The role was not able to be saved.
 permission.permissionsMissing=Permissions {0} missing.
 permission.CannotAssigntDefaultPermissions=Cannot assign default permissions.
+permission.default.contributor.role.none=A person who has no permissions on a newly created dataset. Not recommended for dataverses with human contributors. 
```
Pesky humans.

pdurbin commented on src/main/java/edu/harvard/iq/dataverse/ManagePermissionsPage.java:
```diff
@@ -249,6 +249,49 @@ public void setAuthenticatedUsersContributorRoleAlias(String authenticatedUsersC
+    
+    public String getCustomDefaultContributorRoleName(){
+        if (dvObject instanceof Dataverse && isCustomDefaultConributorRole() ){           
+            return defaultContributorRoleAlias.equals("none") ? "None" : roleService.findCustomRoleByAliasAndOwner(defaultContributorRoleAlias,dvObject.getId()).getName();
```
Should this English word "None" be moved to the bundle so it can be translated?

pdurbin commented on src/main/java/edu/harvard/iq/dataverse/ManagePermissionsPage.java:
```diff
@@ -249,6 +249,49 @@ public void setAuthenticatedUsersContributorRoleAlias(String authenticatedUsersC
+    
+    public String getCustomDefaultContributorRoleAlias(){
+        if (dvObject instanceof Dataverse && isCustomDefaultConributorRole()){
+            return defaultContributorRoleAlias.equals("none") ? "none" : roleService.findCustomRoleByAliasAndOwner(defaultContributorRoleAlias,dvObject.getId()).getAlias();
```
Should this English word "none" be moved to the bundle so it can be translated?

pdurbin commented on src/main/java/edu/harvard/iq/dataverse/ManagePermissionsPage.java:
```diff
@@ -249,6 +249,49 @@ public void setAuthenticatedUsersContributorRoleAlias(String authenticatedUsersC
+    
+    public String getCustomDefaultContributorRoleDescription(){
+        if (dvObject instanceof Dataverse  && isCustomDefaultConributorRole()){
+            return defaultContributorRoleAlias.equals("none") ? BundleUtil.getStringFromBundle("permission.default.contributor.role.none" ) :roleService.findCustomRoleByAliasAndOwner(defaultContributorRoleAlias,dvObject.getId() ).getDescription();
```
Should this English word "none" be moved to the bundle so it can be translated?

pdurbin commented on src/main/java/edu/harvard/iq/dataverse/ManagePermissionsPage.java:
```diff
@@ -249,6 +249,49 @@ public void setAuthenticatedUsersContributorRoleAlias(String authenticatedUsersC
         return defaultContributorRoleAlias;
     }
+    
+    public Boolean isCustomDefaultConributorRole(){
```
Typo: "Conributor"

pdurbin commented on src/main/java/edu/harvard/iq/dataverse/ManagePermissionsPage.java:
```diff
@@ -265,8 +308,8 @@ public void initAccessSettings() {
-
-            defaultContributorRoleAlias = ((Dataverse) dvObject).getDefaultContributorRole().getAlias();
+           
+            defaultContributorRoleAlias = ((Dataverse) dvObject).getDefaultContributorRole() == null ? "none" : ((Dataverse) dvObject).getDefaultContributorRole().getAlias();
```
Should this English word "none" be moved to the bundle so it can be translated?

pdurbin commented on src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java:
```diff
@@ -109,8 +113,10 @@
-//    SystemConfig systemConfig;
+    
+    @EJB
+    DataverseRoleServiceBean roleService;
```
You could use `rolesSvc` from AbstractApiBean instead of injecting a new EJB.

pdurbin commented on src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java:
```diff
@@ -743,6 +749,52 @@ public Response updateGroup(ExplicitGroupDTO groupDto,
+
+        DataverseRole defaultRole;
+        
+        if (roleAlias.equals("none")) {
```
Ok, it's more clear to me now that "none" is a special keyword, not something that appears in the UI. Right? Can we make "none" into a constant? Otherwise we have "none" sprinkled in various places in the code.

pdurbin commented on src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java:
```diff
@@ -743,6 +749,52 @@ public Response updateGroup(ExplicitGroupDTO groupDto,
+        try {
+            Dataverse dv = findDataverseOrDie(dvIdtf);
+            
+            String defaultRoleName = defaultRole == null ? "None" : defaultRole.getName();
```
Convert "None" to a constant?

sekmiller commented:
@pdurbin I addressed all of your code review notes except for the one on Creating Custom Roles documentation. I added an example of the json file at the request of Mike, who was having trouble with testing the UI.  It's not really in scope here. Can we make it a separate ticket?
